frontends/election-manager/src/components/export_ballot_pdfs_button.tsx

```diff
@@ -0,0 +1,375 @@
+import { Buffer } from 'buffer';
+import React, { useCallback, useContext, useState, useMemo } from 'react';
+import styled from 'styled-components';
+import { join } from 'path';
+import { BallotLocales } from '@votingworks/types';
+
+import {
+  assert,
+  usbstick,
+  throwIllegalValue,
+  BALLOT_PDFS_FOLDER,
+  BallotStyleData,
+} from '@votingworks/utils';
+import {
+  Modal,
+  Prose,
+  UsbControllerButton,
+  isAdminAuth,
+  isSuperadminAuth,
+} from '@votingworks/ui';
+import { LogEventId } from '@votingworks/logging';
+import {
+  getBallotArchiveFilename,
+  getBallotPath,
+  getBallotStylesData,
+  sortBallotStyleDataByPrecinct,
+} from '../utils/election';
+
+import { AppContext } from '../contexts/app_context';
+import { HandMarkedPaperBallot } from './hand_marked_paper_ballot';
+import { Button } from './button';
+import { LinkButton } from './link_button';
+import { Loading } from './loading';
+
+import { BallotMode } from '../config/types';
+import { DownloadableArchive } from '../utils/downloadable_archive';
+import { BallotTypeToggle } from './ballot_type_toggle';
+import { BallotModeToggle } from './ballot_mode_toggle';
+import { PrintableArea } from './printable_area';
+import { DEFAULT_LOCALE } from '../config/globals';
+import { generatePdfExportMetadataCsv } from '../utils/generate_pdf_export_metadata_csv';
+
+const { UsbDriveStatus } = usbstick;
+const UsbImage = styled.img`
+  margin-right: auto;
+  margin-left: auto;
+  height: 200px;
+`;
+
+const CenteredOptions = styled.div`
+  text-align: center;
+`;
+
+enum ModalState {
+  BEGIN_EXPORT = 'begin export',
+  GENERATE_FILES = 'generate files',
+  END_EXPORT = 'end export',
+  DONE = 'done',
+  ERROR = 'error',
+}
```

Comment:
Same comment here as earlier about unions of string literals vs. enums. I used a union of string literals for a very similar purpose (modal states) back when we were working on L&A!

https://github.com/votingworks/vxsuite/blob/3df0503a5e13fcca49f3b4036bbb92787025393b/frontends/election-manager/src/screens/print_test_deck_screen.tsx#L333-L337